### PR TITLE
Rename boundary condition plugins, namespaces, functions

### PR DESCRIPTION
**Pull request by gassmoeller**
This PR goes toward resolving #179, and is part of #1090. As discussed in #179 the current directory structure and naming conventions of initial conditions and boundary conditions are somewhat messed up, which makes it harder to understand the code and create new plugins. This PR renames boundary conditions in a consistent way, which unfortunately breaks all existing user plugins for boundary velocities, fluid pressures, and tractions, but it provides a script for sed which pretty reliably converts them into the new naming scheme (it worked for all plugins within master, without manual work).
This PR is compatible with old parameter files, because the subsections are luckily already named in the new scheme.

Maybe we should merge this PR together with a similar PR for the initial conditions (next on my list) so that users only need to update their plugins and parameter files once.

### CODE REVIEW

File: doc/indent [hunk 1]
```diff
@@ -46,4 +46,4 @@ fi
 # with up to 10 in parallel
 echo "--- Indenting all ASPECT header and source files"
 
-find tests include source benchmark cookbooks \( -name '*.cc' -o -name '*.h' \) -print | xargs -n 50 -P 10 astyle --options=doc/astyle.rc
+find tests include source benchmarks cookbooks \( -name '*.cc' -o -name '*.h' \) -print | xargs -n 50 -P 10 astyle --options=doc/astyle.rc
```
Comment (tjhei) [inline doc/indent]: can we get this as a separate PR to merge ASAP?

**Comment by tjhei** [comment]
This looks pretty good. Are all changes to the code done using the included script?

File: doc/indent (hunk 1 above)
Comment (gassmoeller) [inline doc/indent]: Done. See #1450.

**Comment by gassmoeller** [comment]
Yes all code changes in this PR were done by the script (except for the changes in `doc/modules/groups.h`).

File: doc/modules/groups.h [hunk 2]
```diff
@@ -183,6 +183,16 @@
  */
 
 /**
+ * @defgroup BoundaryFluidPressures Describing fluid pressure boundary conditions 
+ * for the fluid velocity field
```
Comment (bangerth) [inline doc/modules/groups.h]: nice job adding this as well.

File: doc/modules/groups.h [hunk 3]
```diff
@@ -183,6 +183,16 @@
...
+ *
+ * A module for the definition of functions and classes that have to do with
+ * describing fluid pressure boundary values for the fluid velocity field.
```
Comment (bangerth) [inline doc/modules/groups.h]: I think this is only used for the melt stuff, right? If so, maybe add a sentence to that effect.

File: doc/update_source_files_to_2.0.0.sed [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# A script for the stream editor sed to update .cc and .h files from the
+# naming scheme used in ASPECT 1.5.0 to ASPECT 2.0.0. This script correctly
+# updated all files within the official development version of ASPECT,
+# but it is not guaranteed to work for all possible names in user plugins.
+# Consequently a backup of files is strongly recommended, and the changes
+# created by this script should be investigated to ensure a correct renaming.
+#
+# Usage for a file named FILENAME: 
+# sed -i -f update_source_files.sed FILENAME
```
Comment (bangerth) [inline doc/update_source_files_to_2.0.0.sed]: can you say what `FILENAME` is supposed to be here? A single source file? Can it contain wildcards?

**Pull request merged**